pkg/cloud/ovirt/providerIDcontroller/providerIDController.go

```diff
@@ -0,0 +1,159 @@
+package providerIDcontroller
+
+import (
+	"context"
+	"fmt"
+
+	ovirtsdk "github.com/ovirt/go-ovirt"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/openshift/cluster-api-provider-ovirt/pkg/cloud/ovirt/clients"
+)
+
+var _ reconcile.Reconciler = &ProviderIDReconciler{}
+
+type ProviderIDReconciler struct {
+	operatorName         string
+	client               client.Client
+	listNodesByFieldFunc func(key, value string) ([]corev1.Node, error)
+	fetchProviderIDFunc  func(string) (string, error)
+	ovirtApi             *ovirtsdk.Connection
+}
+
+func (p *ProviderIDReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	klog.Infof("Reconciling providerID for node %s", request.NamespacedName)
+
+	// Fetch the Node instance
+	node := corev1.Node{}
+	err := p.client.Get(context.Background(), request.NamespacedName, &node)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		klog.Errorf("Error getting node: %v", err)
+		return reconcile.Result{}, fmt.Errorf("error getting node: %v", err)
+	}
+
+	if node.Spec.ProviderID != "" {
+		return reconcile.Result{}, nil
+	}
+
+	klog.Info("spec.ProviderID is empty, fetching from ovirt")
+	id, err := p.fetchProviderIDFunc(node.Name)
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("Failed getting VM from oVirt: %v", err)
```

Comment:
nit: fmt.Errorf nested errors should start with lower case by convention.

Reply:
done